### ISSUE TITLE
Complete `README.md`

### ISSUE THREAD

**Issue by jwaiton** [nu-ZOO/MULE#1]
At the moment, the `README` doesn't describe the full feature set of the repo. This is because (obviously) the full feature set hasn't been implemented but I believe a good `README` (or included docs) should describe the repo concisely and more or less completely, so a list of requirements for the `README` could be a good tool to structure the work the repo requires.

Currently, requirements are:
- [ ] Describe better **What is MULE?**
- [ ] Include relevant explanation for **Usage** of MULE.
- [x] Include **Getting Started**.
   - [x] Add section explaining how to use the executable, and the config files
- [ ] Reassess what further sections are required.

- [x] Add type folder structure for accessing custom types (mostly for typehints)

**Comment by MattZur** [comment]
@jwaiton Would cloning the directory be the simplest method of installation?

**Comment by jwaiton** [comment]
> [@jwaiton](https://github.com/jwaiton) Would cloning the directory be the simplest method of installation?

Yes, that's generally how it should be done. I'll list out how I recall it being done (although some of the steps may be a little wrong, since I'm recalling from memory).

Clone the repo
`git clone https://github.com/nu-ZOO/MULE.git`
`cd MULE`
Add your remote branch
`git remote add MattZur https://github.com/MattZur/MULE.git`
Check that it exists with:
`git remote`
You should see it in the list of remotes. Then fetch and pull it to add it locally.
`git fetch`
`git pull MattZur`
And once you want to start editing it swap to the branch with
`git checkout MattZur/Add-Getting-Started-documentation`
You can check that you're in the right place (and if you have any uncommited changes) with
`git status`

**Comment by MattZur** [comment]
Yes, good. I'll check that with my notes
